### DISCUSSION TITLE
Read all <Import Project="xxx"> from vcxproj file

### DISCUSSION THREAD

**Discussion by shulkx** [3F/MvsSln#52] [Q/A]
I would like to read all import project from vcxproj file
`  <ImportGroup Condition="'$(Configuration)|$(Platform)'=='Debug|x64'" Label="PropertySheets">
    <Import Project="$XXXXXXXX.props" />
  </ImportGroup>`
Is it possible with your library? if yes, how?

**Comment by 3F** [comment]
The *Import* sections in project files for 2.x can be processed with msbuild; For 3.x through IeXod but this is expected to be compatible with msbuild too.

Thus, ~

```csharp
using Sln l = new("<.sln>", SlnItems.Env);

l.Result.Env
    .LoadProjects(l.Result.ProjectItemsConfigs.Where(p => p.project.EpType == ProjectType.Vc))
    .SelectMany(xp => xp.GetImports())
    .Where(x => x.project.EndsWith(".props") && x.project.StartsWith("$"))
    ...
```

**Reply by shulkx** [reply]
Thanks for answering.
I have tried the code and aware that only the loaded project can be obtained with those info by using these codes. Is that right?
I am wrting the C# program which need to analyze the C++ project folder.
I wonder that if it is possible to parser vcxproj file for different C++ project(s) only by passing the full path of sln file.

**Reply by 3F** [reply]
The loading is needed to **evaluate** the xml nodes due to various conditions like your mentioned `"'$(Configuration)|$(Platform)'=='Debug|x64'"` and so on. My answer simply provides the most flexible solution in addition to basic loading of all *minimal* or *all possible* in your environment. I mean, you can just

```csharp
new Sln("<your .sln file>", SlnItems.Env | SlnItems.LoadMinimalDefaultData);
// ... you don't need to configure LoadProjects() anymore
```

Here's [more about `EnvWithMinimalProjects`](https://github.com/3F/MvsSln#did-you-know)

> only by passing the full path of sln file.

It seems to be like above, isn't it?
